instant-distance/src/lib.rs

```diff
@@ -225,74 +246,35 @@ where
             return (
                 Self {
                     ef_search,
-                    zero: Vec::new(),
-                    points: Vec::new(),
-                    layers: Vec::new(),
+                    neighbors: Vec::new(),
+                    meta: Meta::default(),
+                    storage: ContiguousStorage::empty(),
                 },
                 Vec::new(),
             );
         }
 
-        // Determine the number and size of layers.
-
-        let mut sizes = Vec::new();
-        let mut num = points.len();
-        loop {
-            let next = (num as f32 * ml) as usize;
-            if next < M {
-                break;
-            }
-            sizes.push((num - next, num));
-            num = next;
-        }
-        sizes.push((num, num));
-        sizes.reverse();
-        let top = LayerId(sizes.len() - 1);
```

Comment:
It would probably be nice to have a separate commit that moves this into the `types` module, so that it can later morph into the `Meta` type (and in that transition make it easier to review what changed to get there).

Reply:
Ok I'll take this into account when I do another rebase.

Reply:
Done.